hphp/hack/src/typing/typing_attributes.ml

```diff
@@ -63,8 +63,22 @@ let check_implements
       then
         ()
       else
+        let all_user_attribute =
+          let bindings = SMap.bindings SN.UserAttributes.as_map in
```

Comment:
We don't want to offer invalid attributes. If you look at the second value in this map, it says whether an attribute is for classes, for methods, etc.

https://github.com/facebook/hhvm/blob/f3546604cc93d6eea6141346aaf054aa6539dcf7/hphp/hack/src/naming/naming_special_names.ml#L347-L354

You probably want to pass in `attr_interface` and follow the pattern on line 48.